src/cimplog.c

```diff
@@ -0,0 +1,60 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include "cimplog.h"
+
+void __cimplog(const char *module, int level, const char *msg, ...)
```

Comment:
You should make this __cimplog (const char *module, int level, const char *msg, va_list arg_ptr)

Then you can create some flavors on top of it.